src/main.rs

```diff
@@ -87,168 +82,120 @@ async fn floodfill_pt(data: web::Data<AppState>, input: web::Json<UserInputJSON>
-    assert!(node_values_1d_guard.len() == expected_len);
-    //}
+    let expected_len = graph_walk.len() * 32;
+    assert!(node_values_1d.len() == expected_len);
 
     println!(
         "input.new_build_additions.len(): {}",
         input.new_build_additions.len()
     );
+    // TODO Redundant conditional?
     if input.new_build_additions.len() >= 1 {
         for new_build in &input.new_build_additions {
             let value_to_add = new_build[0];
             let index_of_nearest_node = new_build[1];
             let column_to_change = new_build[2];
             let ix = (index_of_nearest_node * 32) + column_to_change;
-            node_values_1d_guard[ix as usize] += value_to_add;
+            node_values_1d[ix as usize] += value_to_add;
         }
         //node_values_1d_guard = add_new_node_values(node_values_1d_guard,&input);
     }
 
-    let time_of_day_ix: usize = get_time_of_day_index(input.trip_start_seconds);
-    let mut model_parameters_each_start = Vec::new();
-    let arc_node_values_1d: Arc<Mutex<Vec<i32>>>;
-    let arc_graph_walk: Arc<Mutex<Vec<SmallVec<[EdgeWalk; 4]>>>>;
-    let arc_graph_pt: Arc<Mutex<Vec<SmallVec<[EdgePT; 4]>>>>;
-    let parallel_res: Vec<(i32, u32, [i64; 32], Vec<u32>, Vec<u16>)>;
-
-    // functionalise or split into two apis
-    if input.year < 2022 {
-        let (node_values_1d, graph_walk, graph_pt, node_values_padding_row_count) =
-            read_files_serial_excluding_travel_time_relationships_and_subpurpose_lookup(input.year);
-
-        println!("Got files read in for {}", input.year);
-        arc_node_values_1d = Arc::new(Mutex::new(node_values_1d));
-        arc_graph_walk = Arc::new(Mutex::new(graph_walk));
-        arc_graph_pt = Arc::new(Mutex::new(graph_pt));
+    let time_of_day_ix = get_time_of_day_index(input.trip_start_seconds);
 
-        let graph_walk_guard = arc_graph_walk.lock().unwrap();
-        let graph_pt_guard = arc_graph_pt.lock().unwrap();
-        let node_values_1d_guard = arc_node_values_1d.lock().unwrap();
+    let count_original_nodes = graph_walk.len() as u32;
 
-        let count_original_nodes: u32 = graph_walk_guard.len() as u32;
-
-        let graph_walk_unguarded = &*graph_walk_guard;
-        let graph_pt_unguarded = &*graph_pt_guard;
-        let node_values_1d_unguarded = &*node_values_1d_guard;
-
-        println!(
-            "Creating tuples to pass to floodfill for {} data",
-            input.year
-        );
-        for i in 0..input.start_nodes_user_input.len() {
-            model_parameters_each_start.push((
-                graph_walk_unguarded,
-                NodeID(input.start_nodes_user_input[i] as u32),
-                node_values_1d_unguarded,
+    println!(
+        "Started running floodfill\ttime_of_day_ix: {}\tNodes count: {}",
+        time_of_day_ix,
+        input.start_nodes_user_input.len()
+    );
+    let now = Instant::now();
+    let indices = (0..input.start_nodes_user_input.len()).collect::<Vec<_>>();
```

Comment:
Stashing all the input to floodfill in a `Vec` just to call `par_iter` is unnecessary. This is still messy (because any function taking 11 arguments is a problem), but less levels of indirection, so it should make later refactoring simpler